compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -71,6 +71,10 @@ object ProtoTypes {
                    |constraint was: ${ctx.typerState.constraint}
                    |constraint now: ${newctx.typerState.constraint}""")
             if result && (ctx.typerState.constraint ne newctx.typerState.constraint) then
+              val tvars = (newctx.typerState.ownedVars -- ctx.typerState.ownedVars).toList
+              inContext(newctx):
+                Inferencing.instantiateSelected(tp, tvars)
+                for tvar <- tvars do if !tvar.isInstantiated then tvar.instantiate(fromBelow = false)
```

Comment:
Does `fromBelow = false` rather than `true` make a difference here?

Reply:
No, it's just following the norm of maximising unconstrained invariant lambda parameters to Any rather than minimising to Nothing.  The goal is to instantiate them with anything, so the lambda can be `.remove`'d from the constraint.

Reply:
Used `Inferencing.instantiateSelected` + this, as `instantiateSelected` is the only current API where you can specify the tvars (and I didn't want to add yet more general purpose ways to instantiate things..)

Reply:
> No, it's just following the norm of maximising unconstrained invariant lambda parameters

Are we certain that they're unconstrained and not constrained by something in `pt`?

Reply:
Changed the approach.